### PR TITLE
WIP: src/components/routes: add component RouteCalendarPanel to log data about active routes

### PR DESCRIPTION
**Pull request by maceksimon**
Add `RouteCalendarPanel` to log data about active routes

* Add component based on q-dialog
* Use panel component in `RoutesCalendar` component
* Refactor logging behaviour into `useLogRoutes` composable
* Add tests
* Add translation strings

---

TODO
- [x] Integrate PR #471
- [x] Finalize panel-trigger behaviour
- [ ] Finalize behaviour when logging multiple routes

### CODE REVIEW

File: src/components/routes/RouteCalendarPanel.vue [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+    });
+
+    // Make props into computed ref so it can be passed as a reactive value.
+    const routes = computed(() => props.routes);
+    // Get panel input state from a composable.
+    const { action, distance, routesCount, transportType, isShownDistance } =
+      useLogRoutes(routes);
+
+    // Determines if save button should be disabled.
+    const isDisabledSave = computed((): boolean => {
```
Comment (tmszi) [inline src/components/routes/RouteCalendarPanel.vue:78]: Better const name `isSaveBtnDisabled`

Comment (maceksimon) [reply src/components/routes/RouteCalendarPanel.vue:78]: Updated.

File: src/composables/useLogRoutes.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  // update panel input data when routes change (user clicks on calendar route).
+  setInputsFromRoute(routes.value);
+  watch(routes, () => {
+    setInputsFromRoute(routes.value);
+  });
+
+  const isShownDistance = computed((): boolean => {
+    return (
+      transportType.value === 'bike' ||
```
Comment (tmszi) [inline src/composables/useLogRoutes.ts:44]: replace `'bike'`string literal with enum, please.

File: src/components/routes/RouteInputDistance.vue [hunk 3]
```diff
@@ -133,7 +133,7 @@ export default defineComponent({
             outlined
             type="number"
             v-model="distance"
-            min="1"
+            min="0"
```
Comment (tmszi) [inline src/components/routes/RouteInputDistance.vue:136]: Mask example https://github.com/quasarframework/quasar/blob/dev/docs/src/examples/QInput/MaskFillReverse.vue

File: src/components/routes/RoutesCalendar.vue [hunk 4]
```diff
@@ -100,9 +108,43 @@ export default defineComponent({
...
+          ) {
+            // Route is already logged - load data.
+            routes.push({ ...day.fromWork });
+          } else {
+            // Route is not logged - create empty route.
+            routes.push({
+              id: '',
+              date: activeRoute.timestamp.date,
+              direction: activeRoute.direction,
+              transport: 'bike',
```
Comment (tmszi) [inline src/components/routes/RoutesCalendar.vue:139]: Use enum please.

File: src/components/routes/RoutesCalendar.vue [hunk 5]
```diff
@@ -180,26 +222,53 @@ export default defineComponent({
...
+
+    /**
+     * Control dialog open state based on selected routes count.
+     */
+    const isOpenPanel = ref<boolean>(false);
+    watch(
+      (): number => activeRoutes.value.length,
+      (length): void => {
+        if (length === 0) {
+          isOpenPanel.value = false;
```
Comment (tmszi) [inline src/components/routes/RoutesCalendar.vue:241]: `isOpenPanel` -> `isPanelOpen`

File: src/components/types/Route.ts [hunk 6]
```diff
@@ -36,3 +39,9 @@ export type RouteCalendarActive = {
 };
 
 export type RouteTab = 'calendar' | 'list' | 'map' | 'app';
```
Comment (tmszi) [inline src/components/types/Route.ts:41]: Use inside `RouteTab` type enum, please. Make it in the new PR, please.

File: src/composables/useLogRoutes.ts [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  /**
+   * Sets the panel input data based on the provided routes.
+   * @param {RouteItem[]} routes - An array of route items.
+   * @return {void}
+   */
+  const setInputsFromRoute = (routes: RouteItem[]): void => {
+    if (routes.length === 1) {
+      action.value = routes[0].inputType || 'input-number';
+      distance.value = routes[0].distance || 0;
+      transportType.value = routes[0].transport || 'bike';
```
Comment (tmszi) [inline src/composables/useLogRoutes.ts:28]: Replace `'bike'` literal string with enum, please.

File: src/composables/useLogRoutes.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    } else {
+      action.value = 'input-number';
+      distance.value = 0;
+      transportType.value = 'bike';
```
Comment (tmszi) [inline src/composables/useLogRoutes.ts:32]: Replace `'bike'` literal string with enum, please.

File: src/components/__tests__/RouteInputDistance.cy.js [hunk 9]
```diff
@@ -246,6 +260,33 @@ function validateZeroTests() {
...
+function valuePassingTests() {
+  it('validates input when 0.5', () => {
+    cy.dataCy(selectorSectionInputNumber)
+      .find('input')
+      .should('be.visible')
+      .and('have.value', 0.5);
+    cy.dataCy(selectorSectionInputNumber).find('input').focus();
+    cy.dataCy(selectorSectionInputNumber).find('input').blur();
+    cy.dataCy(selectorSectionInputNumber)
+      .find(classSelectorMessages)
```
[the remark this answers is not in the document]
Comment (maceksimon) [reply src/components/__tests__/RouteInputDistance.cy.js:272]: No, I have tested multiple approaches including this one. `cy.wait(...)` is prohibited by eslint rules and `timeout` does not ensure awaiting the animation (so the test passes even when if the field is not valid).

File: src/components/routes/RouteInputDistance.vue (hunk 3 above)
Comment (maceksimon) [reply src/components/routes/RouteInputDistance.vue:136]: See https://quasar.dev/vue-components/input#mask
Mask is only available if the type is one of ‘text’ (default), ‘search’, ‘url’, ‘tel’, or ‘password’.
I would prefer `number` type in this case. Do you think we should use input `text` + mask?

File: src/composables/useLogRoutes.ts (hunk 8 above)
Comment (maceksimon) [reply src/composables/useLogRoutes.ts]: Replaced using enum.

File: src/composables/useLogRoutes.ts (hunk 7 above)
Comment (maceksimon) [reply src/composables/useLogRoutes.ts]: Replaced using enum.

File: src/components/routes/RoutesCalendar.vue (hunk 5 above)
Comment (maceksimon) [reply src/components/routes/RoutesCalendar.vue]: Updated.

File: src/components/routes/RoutesCalendar.vue (hunk 4 above)
Comment (maceksimon) [reply src/components/routes/RoutesCalendar.vue]: Replaced using enum.

File: src/composables/useLogRoutes.ts (hunk 2 above)
Comment (maceksimon) [reply src/composables/useLogRoutes.ts]: Replaced using enum.

File: src/components/types/Route.ts (hunk 6 above)
Comment (maceksimon) [reply src/components/types/Route.ts:47]: Updated in PR #515.

File: src/components/routes/RouteInputDistance.vue (hunk 3 above)
Comment (tmszi) [reply src/components/routes/RouteInputDistance.vue:136]: > See https://quasar.dev/vue-components/input#mask Mask is only available if the type is one of ‘text’ (default), ‘search’, ‘url’, ‘tel’, or ‘password’. I would prefer `number` type in this case. Do you think we should use input `text` + mask?

I think yes this example work as expected.

File: src/components/__tests__/RouteInputDistance.cy.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (tmszi) [reply src/components/__tests__/RouteInputDistance.cy.js:272]: > No, I have tested multiple approaches including this one. `cy.wait(...)` is prohibited by eslint rules and `timeout` does not ensure awaiting the animation (so the test passes even when if the field is not valid).

Is this issue really related with waiting for animation? Default Cypress global settings animation waiting is enabled by default https://docs.cypress.io/guides/references/configuration#Actionability.

Did you saw using Promise (with `setTime` call back func) in some Cypress doc or examples?

File: src/components/routes/RouteInputDistance.vue (hunk 3 above)
Comment (tmszi) [reply src/components/routes/RouteInputDistance.vue:136]: Make it in the new PR, please.

**Pull request merged**